### ISSUE TITLE
Feature Request - Add an option to go FullScreen

### ISSUE THREAD

**Issue by bradmartin** [bradmartin/nativescript-videoplayer#14]
Need to add the ability to go Fullscreen with the video.

**Comment by ilyago** [comment]
+1

**Comment by bradmartin** [comment]
Here is a possible solution for now https://github.com/bradmartin/nativescript-videoplayer/issues/28#issuecomment-240914968

I'll add a method to toggle 'Immersive Mode' for Android in the future.

**Comment by bradmartin** [comment]
More difficult than just setting the window flags on Android. I'll try to find time to work on this later, not much free time right now. Definitely open for PRs if someone wants to work on it.

**Comment by lukeramsden** [comment]
+1

**Comment by jbfulgencio** [comment]
Is there any news on this? If not, is there something that I can do to help?

**Comment by bradmartin** [comment]
I haven't had any time lately to work on it. Making it work with the video player is a lot of work. A good alternative might be to have a separate plugin using one of the popular third party android video libs that has done the work of Fullscreen on all the different android APIs. It might be faster to take that approach. However having this plugin support it would still be great I just haven't had time to work on it. Sorry.

**Comment by bnussey** [comment]
Hey @jbfulgencio in our implementation of the plugin, we actually do this manually by making the video itself 100% height and width. You could even animate that.

**Comment by bradmartin** [comment]
I haven't played with the plugin since the major update. That's awesome
@bnussey

On Sat, Jan 14, 2017, 1:20 PM bnussey <notifications@github.com> wrote:

Hey @jbfulgencio <https://github.com/jbfulgencio> in our implementation of
the plugin, we actually do this manually by making the video itself 100%
height and width. You could even animate that.

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub
<https://github.com/bradmartin/nativescript-videoplayer/issues/14#issuecomment-272646481>,
or mute the thread

**Comment by bnussey** [comment]
Yeah I was also looking at Google's Exoplayer and we've been thinking of using that instead of MediaPlayer. Looks like a more advanced API and greater functionality. Will keep you all posted.

**Comment by lukeramsden** [comment]
@bnussey any updates to that?

**Comment by christian-kolb** [comment]
@bradmartin Is there a function to call to be able to trigger the same fullscreen mode that is triggered when the controls are shown? I looked through all links posted but they are all referring to android.

**Comment by sfaessler** [comment]
+1 any news? Would love to have videos in fullscreen :)

**Comment by c1ngular** [comment]
+1 , people really like to see a video  in full screen :)

**Comment by bradmartin** [comment]
I have no time to work on this right now. If anyone wants to work on it and needs guidance I can chat from time to time but I'm too busy to put in the work to get the functionality working smoothly. If you absolutely need the fullscreen option in your NS app right now, the exoplayer plugin is a perfect solution.

**Comment by souly1** [comment]
Similar to @bnussey, I'm currently working on a fullscreen ability and basically open a nativescript angular modal window with a new video player in it.
so far looks great (checked only on android emulator)
Using (Screen Orientation)[https://github.com/VideoSpike/nativescript-screen-orientation] plugin to switch back a forth if required and the modal window automatically closes when back button pressed calls a callback function.

P.S if using this solution don't forget to 'freeze' the embedded player when opening the full screen one as you don't want it to keep using bandwidth in the background

**Comment by fransyozef** [comment]
+1

**Comment by ableschulze** [comment]
I want it too. Any news ?
